tutorials/haskell/gui-application/tutorial.md

```diff
@@ -0,0 +1,825 @@
+interface for better control of layout. We will be using the following
+functions from its API:
+
+```haskell
+-- | Creates a 'Grid'.
+gridNew :: IO Grid
+
+-- | Sets whether all rows of grid will have the same height.
+gridSetRowHomogeneous :: GridClass self
+  => self              -- ^ The grid
+  -> Bool              -- ^ 'True' to make rows homogeneous
+  -> IO ()
+
+-- | Adds a widget to the grid. The position of child is determined by left
+-- and top. The number of “cells” that child will occupy is determined by
+-- width and height.
+gridAttach :: (GridClass self, WidgetClass child)
+  => self    -- ^ The grid
+  -> child   -- ^ The widget to add
+  -> Int     -- ^ The column number of to attach the left side of child to
+  -> Int     -- ^ The row number to attach the top side of child to
+  -> Int     -- ^ Width — the number of columns that child will span
+  -> Int     -- ^ Height — the number of rows that child will span
+  -> IO ()
+```
+
+`gridNew` and `gridSetRowHomogeneous` should be self-explanatory.
+`gridAttach` allows to insert widgets into the grid controlling their
+position and size. This is very handy for our calculator application, let's
+use it:
+
+```haskell
+main :: IO ()
+main = do
+  …
+  grid <- gridNew                  -- (1)
+  gridSetRowHomogeneous grid True  -- (2)
+  let attach x y w h item = gridAttach grid item x y w h -- (3)
+  attach 0 0 5 1 display           -- (4)
+  mkBtn "MC"  >>= attach 0 1 1 1   -- (5)
+  mkBtn "MR"  >>= attach 1 1 1 1
+  mkBtn "MS"  >>= attach 2 1 1 1
+  mkBtn "M+"  >>= attach 3 1 1 1
+  mkBtn "M–"  >>= attach 4 1 1 1
+  mkBtn "←"   >>= attach 0 2 1 1
+  mkBtn "CE"  >>= attach 1 2 1 1
+  mkBtn "C"   >>= attach 2 2 1 1
+  mkBtn "±"   >>= attach 3 2 1 1
+  mkBtn "√"   >>= attach 4 2 1 1
+  mkBtn "7"   >>= attach 0 3 1 1
+  mkBtn "8"   >>= attach 1 3 1 1
+  mkBtn "9"   >>= attach 2 3 1 1
+  mkBtn "÷"   >>= attach 3 3 1 1
+  mkBtn "%"   >>= attach 4 3 1 1
+  mkBtn "4"   >>= attach 0 4 1 1
+  mkBtn "5"   >>= attach 1 4 1 1
+  mkBtn "6"   >>= attach 2 4 1 1
+  mkBtn "*"   >>= attach 3 4 1 1
+  mkBtn "1/x" >>= attach 4 4 1 1
+  mkBtn "1"   >>= attach 0 5 1 1
+  mkBtn "2"   >>= attach 1 5 1 1
+  mkBtn "3"   >>= attach 2 5 1 1
+  mkBtn "–"   >>= attach 3 5 1 1
+  mkBtn "="   >>= attach 4 5 1 2
+  mkBtn "0"   >>= attach 0 6 2 1
+  mkBtn "."   >>= attach 2 6 1 1
+  mkBtn "+"   >>= attach 3 6 1 1
+  containerAdd window grid         -- (6)
+  …
+```
+
+1. `gridNew` creates a new grid.
+
+2. `gridSetRowHomogeneous grid True` makes every row have equal height.
+
+3. Here we define the `attach` helper function. It attaches given widget to
+   our `grid`. The argument order of this function helps to use it with
+   `(>>=)`.
+
+4. We attach `display` we created previously to the `grid`. It will occupy
+   the entire top row.
+
+5. Here we use combination of `mkBtn` helper and `attach` to quickly create
+   buttons and place them on the grid. I'll show `mkBtn` is a moment.
```

Comment:
@mrkkrp 

- "Here we use a combination of **the** `mkBtn` helper and `attach` to quickly create buttons and place them on the grid".
- Instead of "I'll show `mkBtn` is a moment", is it "I'll show **what** `mkBtn` is **in** a moment"?